brody_lab_to_nwb/msortedsortinginterface.py

```diff
@@ -0,0 +1,39 @@
+"""Authors: Cody Baker."""
+import numpy as np
+
+from h5py import File
+from nwb_conversion_tools.datainterfaces.ecephys.basesortingextractorinterface import BaseSortingExtractorInterface
+
+from .msortedsortingextractor import MSortedSortingExtractor
+
+
+class MSortedSortingInterface(BaseSortingExtractorInterface):
+    """Conversion class for the pre-sorted data corresponding to the Neuralynx format for the Brody lab."""
+
+    SX = MSortedSortingExtractor
+
+    @classmethod
+    def get_source_schema(cls):
+        source_schema = dict(
+            required=["file_path"],
+            properties=dict(
+                file_path=dict(
+                    type="string",
+                    format="file",
+                    description="Path to .mat file containing processed data."
+                )
+            ),
+            type="object",
+            additionalProperties=False
+        )
+        return source_schema
+
+    def __init__(self, **source_data):
+        processed_file_path = source_data["file_path"]
+        mat_file = File(processed_file_path, mode="r")
+        sf = mat_file["Msorted"]["sampling_frequency"][()][0][0]
```

Comment:
again, why fetch it all and then take the first element?

Reply:
Fixed!